src/main.cpp

```diff
@@ -903,15 +965,15 @@ int main ( int argc, char** argv )
 
                 // show dialog
```

Comment:
Why isn't the exit-code variable here?

Reply:
Since here main always returns 0 and we only did set an exit code with `exit(1)`, but in the new situation we wanna do the clean up, so we don't wanna use `exit()`, but we will return from main normally, so main must return the exit code.